src/Core/src/HotReload/HotReloadHelper.cs

```diff
@@ -0,0 +1,154 @@
+				TransferState(view, newView);
+				return newView;
+			}
+			catch (MissingMethodException ex)
+			{
+				Debug.WriteLine("You are using trying to HotReload a view that requires Parameters. Please call `HotReloadHelper.Register(this, params);` in the constructor;");
+				throw ex;
+			}
+
+		}
+
+		static void TransferState(IHotReloadableView oldView, IView newView)
+		{
+
+			oldView.TransferState(newView);
+		}
+
+		static internal readonly WeakList<IHotReloadableView> ActiveViews = new WeakList<IHotReloadableView>();
+		static Dictionary<string, Type> replacedViews = new Dictionary<string, Type>();
+		static Dictionary<IHotReloadableView, object[]> currentViews = new Dictionary<IHotReloadableView, object[]>();
+		static Dictionary<string, List<KeyValuePair<Type, Type>>> replacedHandlers = new Dictionary<string, List<KeyValuePair<Type, Type>>>();
+		public static void RegisterReplacedView(string oldViewType, Type newViewType)
+		{
+			if (!IsEnabled)
+				return;
+
+			Console.WriteLine($"{oldViewType} - {newViewType}");
+
+			if (typeof(IHotReloadableView).IsAssignableFrom(newViewType))
+				replacedViews[oldViewType] = newViewType;
+
+			if (typeof(IViewHandler).IsAssignableFrom(newViewType))
+			{
+				if (replacedHandlers.TryGetValue(oldViewType, out var vTypes))
+				{
+					foreach (var vType in vTypes)
+						RegisterHandler(vType, newViewType);
+					return;
+				}
+
+				_ = HandlerService ?? throw new ArgumentNullException(nameof(HandlerService));
+				var assemblies = System.AppDomain.CurrentDomain.GetAssemblies();
+				var t = assemblies.Select(x => x.GetType(oldViewType)).FirstOrDefault(x => x != null);
+
+				var views = HandlerService!.Where(x => x.ImplementationType == t).Select(x => new KeyValuePair<Type,Type>(x.ServiceType, x.ImplementationType!)).ToList();
+
+
+				replacedHandlers[oldViewType] = views.ToList();
+				foreach (var h in views)
+				{
+					RegisterHandler(h, newViewType);
+				}
+			}
+			//Call static init if it exists on new classes!
+			var staticInit = newViewType.GetMethod("Init", System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.Public);
+			staticInit?.Invoke(null, null);
+		}
+
+
+		static void RegisterHandler(KeyValuePair<Type, Type> pair, Type newHandler)
+		{
+			_ = HandlerService ?? throw new ArgumentNullException(nameof(HandlerService));
+			var view = pair.Key;
+			var newType = newHandler;
+			if (pair.Value.IsGenericType)
+				newType = pair.Value.GetGenericTypeDefinition().MakeGenericType(newHandler);
+			HandlerService.AddTransient(view, newType);
+		}
+
+		public static void TriggerReload()
+		{
+			List<IHotReloadableView>? roots = null;
+			while (roots == null)
+			{
+				try
+				{
+					//roots = View.ActiveViews.Where(x => (x.Parent is CometApp) || (x.Parent == null && !(x is CometApp) )).ToList();
+					roots = ActiveViews.Where(x => x.Parent == null).ToList();
+				}
+				catch
+				{
+					//Sometimes we get list changed exception.
```

Comment:
Should we throw here ?

Reply:
no, we shouldn't. unless we're sure this will be invoked in a try/catch